Android/app/src/main/java/com/example/instagramclone/Adapters/ViewHolders/FeedPlaceHolder.java

```diff
@@ -0,0 +1,221 @@
+
+            }
+        });
+    }
+
+    private void addLikeToImage(String feedID) {
+        Call<FeedItemModel> call = feedApi.likePost("/api/posts/like/" + feedID, getToken());
+        call.enqueue(new Callback<FeedItemModel>() {
+            @Override
+            public void onResponse(Call<FeedItemModel> call, Response<FeedItemModel> response) {
+                setLikedIcon();
+            }
+
+            @Override
+            public void onFailure(Call<FeedItemModel> call, Throwable t) {
+
+            }
+        });
+    }
+
+    private void removeLikeFromImage(String feedID) {
+        Call<FeedItemModel> call = feedApi.unlikePost("/api/posts/unllike/" + feedID, getToken());
+        call.enqueue(new Callback<FeedItemModel>() {
+            @Override
+            public void onResponse(Call<FeedItemModel> call, Response<FeedItemModel> response) {
+                setUnlikeImage();
+            }
+
+            @Override
+            public void onFailure(Call<FeedItemModel> call, Throwable t) {
+
+            }
+        });
+    }
+
+    private void setUnlikeImage() {
+        int n = itemModel.getLikeCount() + 1;
+        itemModel.setLikeCount(n);
+        like.setVisibility(View.VISIBLE);
+        liked.setVisibility(View.GONE);
+    }
+
+    private void setLikedIcon() {
+        like.setVisibility(View.GONE);
+        liked.setVisibility(View.VISIBLE);
+
+    }
+
+    private void showPopupMenu(Context context) {
+        Log.d("FeedPlaceHolder", "CE GA OTVORI VALJDA");
+        PopupMenu popupMenu = new PopupMenu(this.context, options);
+        popupMenu.getMenuInflater().inflate(R.menu.feed_options_menu, popupMenu.getMenu());
+        popupMenu.getMenu().getItem(0).setOnMenuItemClickListener(item -> {
+            showDeleteConfirmDialog(this.context);
+            return true;
+        });
+        popupMenu.show();
+    }
+
+    private void showDeleteConfirmDialog(Context context) {
+        AlertDialog dialog = new AlertDialog.Builder(context)
+                .setTitle("Delete this post")
+                .setMessage("Are you sure you want to delete this post?")
+                .setPositiveButton("Confirm", ((dialogInterface, i) -> deletePost()))
+                .setNegativeButton("Cancel", (dialogInterface, i) -> dialogInterface.dismiss())
+                .create();
+
+        dialog.show();
+    }
+
+    private void deletePost() {
+        Call<ResponseBody> call = feedApi.deletePost("/api/posts/delete/" + itemModel.getFeedUUID(), getToken());
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                //todo add snackbar
+            }
+
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+                t.printStackTrace();
+            }
+        });
+    }
+
+    private String getToken() {
```

Comment:
I feel like this method is duplicated too many times, is there any better way of handling that?